staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -86,6 +86,12 @@ type Helper struct {
 
 	// OnPodDeletedOrEvicted is called when a pod is evicted/deleted; for printing progress output
 	OnPodDeletedOrEvicted func(pod *corev1.Pod, usingEviction bool)
+
+	// OnPodDeletionOrEvictionFailed is called when a pod is eviction/deletetion is failed; for printing progress output
+	OnPodDeletionOrEvictionFailed func(pod *corev1.Pod, usingEviction bool)
+
+	// OnPodDeletionOrEvictionStarted is called when a pod eviction/deletion is started; for printing progress output
+	OnPodDeletionOrEvictionStarted func(pod *corev1.Pod, usingEviction bool)
```

Comment:
I was thinking about it a bit more and I'm not a fan of having multiple methods for the same thing, but I also recognize the limitation of the current callbacks. I was thinking a bit about this and I came up with this:
1. Deprecated the current `OnPodDeletedOrEvicted`.
2. Introduce `OnPodDeletetionOrEvictionStarted` like in your example and `OnPodDeletionOrEvictionFinished` with `error` as the 3rd parameter to distinguish success from error.

wdyt?

Reply:
I agree, I have made the changes kindly check.